dotCMS/src/main/java/com/dotcms/rest/api/v1/container/ContainerResource.java

```diff
@@ -92,24 +120,244 @@ public ContainerResource(final WebResource webResource,
+    @Path("/{containerId}/content/{contentletId}")
+    public final Response containerContent(@Context final HttpServletRequest req, @Context final HttpServletResponse res,
+            @PathParam("containerId") final String containerId, @PathParam("contentletId") final String contentletId)
+            throws DotDataException, DotSecurityException, ParseErrorException, MethodInvocationException,
+            ResourceNotFoundException, IOException, IllegalAccessException, InstantiationException, InvocationTargetException,
+            NoSuchMethodException {
+
+        final InitDataObject initData = webResource.init(true, req, true);
+        final User user = initData.getUser();
+        PageMode mode = PageMode.get(req);
+        Language id = WebAPILocator.getLanguageWebAPI().getLanguage(req);
+
+
+        ShortyId contentShorty = APILocator.getShortyAPI().getShorty(contentletId).orElseGet(() -> {
+            throw new ResourceNotFoundException("Can't find contentlet:" + contentletId);
+        });
+
+
+
+        final Contentlet contentlet = (contentShorty.subType == ShortType.CONTENTLET)
+                ? APILocator.getContentletAPI().find(contentShorty.longId, user, mode.showLive)
+                : APILocator.getContentletAPI().findContentletByIdentifier(contentShorty.longId, mode == PageMode.ANON,
+                        id.getId(), user, mode == PageMode.ANON);
+
+        if (contentlet.getContentType().baseType() == BaseContentType.WIDGET) {
+            Map<String, String> response = new HashMap<>();
+            response.put("render", WidgetResource.parseWidget(req, res, contentlet));
+
+            return Response.ok(response).build();
+
+
+        }
+
+        ShortyId containerShorty = APILocator.getShortyAPI().getShorty(containerId).orElseGet(() -> {
+            throw new ResourceNotFoundException("Can't find Container:" + containerId);
+        });
+
+        Container container = (containerShorty.subType == ShortType.CONTAINER)
+                ? APILocator.getContainerAPI().find(containerId, user, mode == PageMode.ANON)
+                : (mode.showLive)
+                        ? (Container) APILocator.getVersionableAPI().findLiveVersion(containerShorty.longId, user,
+                                mode == PageMode.ANON)
+                        : (Container) APILocator.getVersionableAPI().findWorkingVersion(containerShorty.longId, user,
+                                mode == PageMode.ANON);
+
+
+        Identifier containerIdentifier = APILocator.getIdentifierAPI().find(container);
+
+        org.apache.velocity.context.Context context = VelocityUtil.getWebContext(req, res);
+        context.remove("EDIT_MODE");
+
+
+
+        context.put("contentletList" + container.getIdentifier(), Lists.newArrayList(contentlet.getIdentifier()));
+        StringWriter out = new StringWriter();
+
+        final ContainerStructure cStruct = APILocator.getContainerAPI().getContainerStructures(container).stream()
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Remove this useless assignment to local variable "cStruct". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1854)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Remove this unused "cStruct" local variable. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1481)